src/main/java/io/xpire/commons/util/StringUtil.java

```diff
@@ -97,4 +103,97 @@ public static String convertToSentenceCase(String string) {
+     * If returns 0, the strings are same.
+     * If returns 1, that means either a character is added, removed, replaced or swapped.
+     * @param source the first string
+     * @param target the second string
+     * @return The Levenshtein Distance between the two strings.
+     */
+    public static int computeDistance (String source, String target) {
+        if (source == null || target == null) {
+            throw new IllegalArgumentException("Parameter must not be null");
+        }
+        int sourceLength = source.length();
+        int targetLength = target.length();
+        if (sourceLength == 0) {
+            return targetLength;
+        }
+        if (targetLength == 0) {
+            return sourceLength;
+        }
+        int[][] dist = new int[sourceLength + 1][targetLength + 1];
+        for (int i = 0; i < sourceLength + 1; i++) {
+            dist[i][0] = i;
+        }
+        for (int j = 0; j < targetLength + 1; j++) {
+            dist[0][j] = j;
+        }
+        for (int i = 1; i < sourceLength + 1; i++) {
+            for (int j = 1; j < targetLength + 1; j++) {
+                int cost = source.charAt(i - 1) == target.charAt(j - 1) ? 0 : 1;
+                //minimum cost between insertion, deletion, replacement
+                dist[i][j] = Math.min(Math.min(dist[i - 1][j] + 1, dist[i][j - 1] + 1), dist[i - 1][j - 1] + cost);
+                //check for transpositions
+                if (i > 1 && j > 1
+                        && source.charAt(i - 1) == target.charAt(j - 2)
+                        && source.charAt(i - 2) == target.charAt(j - 1)) {
+                    dist[i][j] = Math.min(dist[i][j], dist[i - 2][j - 2] + cost);
+                }
+            }
+        }
+        return dist[sourceLength][targetLength];
+        //@@author
+    }
+
+    /**
+     * Returns suggestions of alternatives for an invalid word entered.
+     * @param invalidWord The invalid word entered.
+     * @param set The set of alternative words.
+     * @param limit The maximum degree of differences between words compared which is accepted.
+     * @return Suggestions that are the most appropriate replacements for the word entered.
+     */
+    public static String getSuggestions(String invalidWord, Set<String> set, int limit) {
+        StringBuilder matches = new StringBuilder();
+        TreeMap<Integer, TreeSet<String>> allMatches = new TreeMap<>();
+        for (String s : set) {
+            if (s.startsWith(invalidWord)) {
+                allMatches.putIfAbsent(-1, new TreeSet<>());
+                allMatches.get(-1).add(s);
+            } else if (!allMatches.containsValue(s)) {
+                int i = StringUtil.computeDistance(s.toLowerCase(), invalidWord.toLowerCase());
+                if (i <= limit) {
+                    allMatches.putIfAbsent(i, new TreeSet<>());
+                    allMatches.get(i).add(s);
+                    System.out.println(s);
```

Comment:
Is this println for debugging purpose? Maybe can remove if no longer needed.

Reply:
Noted, will delete it.